Drivers/Periphs/Src/radio.c

```diff
@@ -16,21 +16,24 @@
 
 static QueueHandle_t EthernetQ; // information will be put on this and all you do is trasmit the date that you receive.
 static struct sockaddr_in sLocalAddr;
-static struct linger soLinger = {.l_onoff = true, .l_linger = 0};
 static int servsocket;
+static struct linger soLinger = {.l_onoff = true, .l_linger = 0};
 static uint32_t EthDroppedMessages = 0;    // for debugging purposes
 extern int errno;
 
-/** Ethernet ConnectToServer
- * @brief Waits until server connection is established - blocking
+/** Ethernet waitForClient
+ * @brief Waits until a client is established - blocking funciton that waits until a client is established
  */
-static void Ethernet_ConnectToServer() {
-    while (servsocket < 0) {
-        do {
-            servsocket = lwip_socket(AF_INET, SOCK_STREAM, IPPROTO_TCP);
-        } while (servsocket < 0);
+void Ethernet_WaitForClient(){
+    if (servsocket >= 0) return;
 
-        debugprintf("servsocket %d\n\r", servsocket);
+    struct sockaddr_in client_addr;
+    int addrlen = sizeof(client_addr);
+    while (servsocket < 0) {
+        do{
+            servsocket = lwip_accept(servsocket, (struct sockaddr *)&client_addr, (socklen_t *)&addrlen);
+        }
+        while (servsocket < 0);
```

Comment:
make sure that if lwip_accept fails that immediately retrying a connection is ok.
might need to destroy and recreate/rebind the socket